src/crypto/CHIPCryptoPALOpenSSL.cpp

```diff
@@ -132,6 +132,18 @@ CHIP_ERROR AES_CCM_encrypt(const uint8_t * plaintext, size_t plaintext_length, c
     int result               = 1;
     const EVP_CIPHER * type  = nullptr;
 
+    const uint8_t * empty_plaintext = Uint8::from_const_char("");
+
+    if (plaintext_length != 0)
+    {
+        VerifyOrExit(plaintext != nullptr, error = CHIP_ERROR_INVALID_ARGUMENT);
+        VerifyOrExit(ciphertext != nullptr, error = CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
This should always be true: ciphertext cannot be null.

Reply:
Resolving after slack conversation with @tcarmelveilleux . The nullptr for cipher text works if plain text length is 0. The unit tests are covering this condition.